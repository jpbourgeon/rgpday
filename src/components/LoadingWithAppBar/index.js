import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import CircularProgress from '@material-ui/core/CircularProgress'
import AppBar from '../AppBar'

const styles = theme => ({
  layout: {
    width: 'auto',
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(1100 + theme.spacing.unit * 3 * 2)]: {
      width: 1100,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  container: {
    position: 'fixed',
    top: 0,
    height: '100vh'
  },
  item: {
    textAlign: 'center'
  }
})

function Loading (props) {
  const { classes } = props

  if (props.pastDelay) {
    return (
      <div className={classes.layout}>
        <AppBar />
        <main>

          <Grid container alignContent='center' alignItems='center' className={classes.container}>
            <Grid item xs={12} className={classes.item}>
              <CircularProgress color='secondary' />
            </Grid>
          </Grid>
        </main>
      </div>
    )
  } else {
    return null
  }
}

Loading.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(Loading)
