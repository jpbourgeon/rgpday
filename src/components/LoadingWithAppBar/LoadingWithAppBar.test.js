/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import LoadingWithAppBar from './'
import jssSerializer from '../../../utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The LoadingWithAppBar component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<LoadingWithAppBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
