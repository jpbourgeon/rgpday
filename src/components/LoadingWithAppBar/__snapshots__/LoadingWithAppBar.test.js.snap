// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LoadingWithAppBar component should render correctly 1`] = `null`;
